ironfish/src/rpc/routes/chain/snapshotChain.ts

```diff
@@ -0,0 +1,101 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import bufio from 'bufio'
+import * as yup from 'yup'
+import { Assert } from '../../../assert'
+import { getBlockSize, writeBlock } from '../../../network/utils/block'
+import { BlockchainUtils } from '../../../utils/blockchain'
+import { ApiNamespace, router } from '../router'
+
+const MAX_BLOCKS_PER_SNAPSHOT_CHUNK = 1000
+
+export type SnapshotChainStreamRequest =
+  | {
+      maxBlocksPerChunk?: number | null
+    }
+  | undefined
+
+export type SnapshotChainStreamResponse = {
+  start: number
+  stop: number
+  seq?: number
+  buffer?: Buffer
+}
+
+export const SnapshotChainStreamRequestSchema: yup.ObjectSchema<SnapshotChainStreamRequest> =
+  yup
+    .object({
+      maxBlocksPerChunk: yup.number().nullable().optional(),
+    })
+    .optional()
+
+export const SnapshotChainStreamResponseSchema: yup.ObjectSchema<SnapshotChainStreamResponse> =
+  yup
+    .object({
+      start: yup.number().defined(),
+      stop: yup.number().defined(),
+      seq: yup.number().optional(),
+      buffer: yup.mixed<Buffer>().optional(),
+    })
+    .defined()
+
+router.register<typeof SnapshotChainStreamRequestSchema, SnapshotChainStreamResponse>(
```

Comment:
Maybe a future cleanup could be combining this with `exportChainStream`. The only difference seems to be the serialization format. But they're different enough that makes sense to have them separate for this project Just a thought going forward

Reply:
Agreed, Jason and I had talked about this out-of-band and we'll probably change it at some point and put them together, but we'll wait for now as there are some users that do actively use `exportChainStream`.

Reply:
This is a good idea for the future, but yeah, since we have existing consumers I don't think we have to worry about it for this PR.